flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/batch/DynamicPartitionPruningRule.java

```diff
@@ -189,6 +195,28 @@ protected boolean doMatches(
         return !acceptedFieldIndices.isEmpty();
     }
 
+    /** Returns true if the source is FLIP-27 source, else false. */
+    private static boolean isNewSource(ScanTableSource scanTableSource) {
+        ScanTableSource.ScanRuntimeProvider provider =
+                scanTableSource.getScanRuntimeProvider(ScanRuntimeProviderContext.INSTANCE);
+        if (provider instanceof SourceProvider) {
+            return true;
+        } else if (provider instanceof TransformationScanProvider) {
+            Transformation<?> transformation =
+                    ((TransformationScanProvider) provider)
+                            .createTransformation(name -> Optional.empty());
+            return transformation instanceof SourceTransformation;
+        } else if (provider instanceof DataStreamScanProvider) {
+            // Suppose DataStreamScanProvider of sources that support dynamic filtering will use new
```

Comment:
It is better to emphasize here that this situation will not generate wrong result because it's handled when BatchTableSourceScan is translated to Transformation.  The only effect is physical plan and exec node plan have dpp node, but it does not work in runtime.

Reply:
I will update the comments in my local when merging